packages/angular_devkit/build_optimizer/src/build-optimizer/build-optimizer.ts

```diff
@@ -94,39 +92,34 @@ export function buildOptimizer(options: BuildOptimizerOptions): TransformJavascr
     };
   }
 
-  let selectedGetScrubFileTransformer = getScrubFileTransformer;
-
-  if (
-    isAngularCoreFile === true ||
-    (isAngularCoreFile === undefined && originalFilePath && isKnownCoreFile(originalFilePath))
-  ) {
-    selectedGetScrubFileTransformer = getScrubFileTransformerForCore;
+  if (isAngularCoreFile === undefined) {
+    isAngularCoreFile = !!originalFilePath && isKnownCoreFile(originalFilePath);
   }
 
+  const hasSafeSideEffects = originalFilePath && isKnownSideEffectFree(originalFilePath);
+
   // Determine which transforms to apply.
   const getTransforms: TransformerFactoryCreator[] = [];
 
   let typeCheck = false;
-  if (options.isSideEffectFree || originalFilePath && isKnownSideEffectFree(originalFilePath)) {
+  if (hasSafeSideEffects) {
+    // Angular modules have known safe side effects
     getTransforms.push(
       // getPrefixFunctionsTransformer is rather dangerous, apply only to known pure es5 modules.
       // It will mark both `require()` calls and `console.log(stuff)` as pure.
       // We only apply it to modules known to be side effect free, since we know they are safe.
-      // getPrefixFunctionsTransformer needs to be before getFoldFileTransformer.
       getPrefixFunctionsTransformer,
```

Comment:
Do we know for which cases the prefix function transformer is needed in Angular?

Reply:
View Engine factory files need it and were already included in the safe side effect check.
Additionally, there are a variety call/new expressions in `@angular/core` involving the following:
```
[
  "ComponentFactoryResolver$1",
  "InjectionToken",
  "KeyRegistry",
  "Optional",
  "ObservableStrategy",
  "PromiseStrategy",
  "ReflectionCapabilities",
  "Reflector",
  "Version",
  "core_merge",
  "getClosureSafeProperty",
  "makeDecorator",
  "makeParamDecorator",
  "makePropDecorator",
  "tagSet",
  "tokenKey",
]
```